source/_posts/2018-01-25-Making-a-presentation-app-with-Google-Nearby-Messages-and-Firebase.md

```diff
@@ -0,0 +1,175 @@
+---
+title:  Making a Presentation Application with Google Nearby Messages & Firebase
+date: 2018-01-05 13:00:00
+tags: ios,google,firebase,googlenearbymessages,presentations
+authorIds:
+- anll
+categories:
+- iOS
+---
+
+Here at Nodes, we regularly run events and give presentations. We wanted to use technology to make these more accessible and enhance these events for people who attend. Smooth out the event process entirely the only way we know how, making applications!
+
+Although we have many ideas for what this application could behold, I decided to focus on the presentation process and in particular, presenting slideshows. I set out to make an application that could display slideshows and "broadcast" them to other local devices running the same application, while making the process as simple for a user as possible. I have developed a prototype application that allows users to give presentations while letting other users view the same presentation on their own devices, while it automatically follows along at real time with the presenter. I’m going to explain how this was made possible by using two mobile platforms from Google, [Firebase](https://firebase.google.com/) & [Nearby Messages](https://developers.google.com/nearby/).
+
+The application currently works with PDFs; using a `UIImage` extension, it takes PDF files and turns pages into images. I then use a `UIPageViewController` to create the required number of ViewControllers and display each pdf page in a separate view controller allowing the user to flick through them. In this blog, I want to focus more on the Google platforms. The project including this code is available here. [github/nodes-presenter-iOS](https://github.com/nodes-projects/nodes-presenter-ios/tree/PresentationApp_Development).
```

Comment:
I'm not sure this should be public / open-source. This app could be a product that Nodes can brand and sell to clients, so it may be worth using a gist to post the pieces of code you want to talk about, or just remove the references to the GitHub repo?

Reply:
Ah ok, I did post code snippets further down. Its just I didn't want to go in to more detail about that side of the app so just put the link to the project there. I will take out this link